server/models/base/Model.ts

```diff
@@ -30,10 +25,20 @@ import Logger from "@server/logging/Logger";
 import { Replace, APIContext } from "@server/types";
 import { getChangsetSkipped } from "../decorators/Changeset";
 
-export type EventOverride = {
+type EventOverrideOptions = {
+  /** Override the default event name. */
   name?: string;
 };
 
+type EventOptions = EventOverrideOptions & {
+  /**
+   * Whether to publish event to the job queue. Defaults to true when using any `withCtx` methods.
+   */
+  publish: boolean;
```

Comment:
I think "publish" is just a little overloaded in Outline already in regards to publishing documents. It will also _write_ events to the db – maybe just "create" is ok?